File: longest_common_prefix/longest_prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) < 1:
            return ""
        ss = ""
        for i in range(len(strs[0])):
            if self.isPrex(strs[0][i], strs, i):
                ss = ss + strs[0][i]
            else:
                break
            
        return ss
    
    def isPrex(self, c, strs, pos):
        for s in strs:
            if len(s) <= pos or c != s[pos]:
                return False    
        return True

File: longest_common_prefix/longest_prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.size() < 1)
            return "";
        string ss = "";
        for(int i = 0; i < strs[0].length(); i++) {
            if(isPrex(strs[0][i], strs, i))
                ss.push_back(strs[0][i]);
            else
                break;
        }
        
        return ss;
    }
    
    bool isPrex(char c, vector<string> strs, int pos) {
        for(int i = 0; i < strs.size(); i++) {
            if(strs[i].size() < pos) 
                break;
            if(c != strs[i][pos])
                return false;
        }
        return true;
    }
};
